test/e2e/spec/shared_spec.rb

```diff
@@ -249,9 +248,246 @@
+          expect(res.to_s).to include addr_prefix
+        end
+
+        matrix_h.each do |role, addr_prefix|
+          id = [*0..100000].sample
+          res = SHARED.keys.get_public_key(pending_wid, role, id, {hash: true})
+          expect(res).to be_correct_and_respond 200
+          expect(res.to_s).to include addr_prefix
+        end
+      end
+
+      it "Get public key - active wallet from mnemonics" do
+        pending "utxo_internal returns empty response"
+        m24 = mnemonic_sentence(24)
+        active_wid = create_pending_shared_wallet(m24, '11H', 'self')
+
+        matrix.each do |role, addr_prefix|
+          id = [*0..100000].sample
+          res = SHARED.keys.get_public_key(active_wid, role, id)
+          expect(res).to be_correct_and_respond 200
+          expect(res.to_s).to include addr_prefix
+        end
+
+        matrix_h.each do |role, addr_prefix|
+          id = [*0..100000].sample
+          res = SHARED.keys.get_public_key(active_wid, role, id, {hash: true})
+          expect(res).to be_correct_and_respond 200
+          expect(res.to_s).to include addr_prefix
+        end
+      end
 
+      it "Get public key - active wallet from acc pub key" do
+        pending "utxo_internal returns empty response"
+        m24 = mnemonic_sentence(24)
+        acc_ix = '0H'
+        acc_xpub = cardano_address_get_acc_xpub(m24, "1854H/1815H/#{acc_ix}")
+        active_wid = create_active_shared_wallet(acc_xpub, acc_ix, "self")
+
+        matrix.each do |role, addr_prefix|
+          id = [*0..100000].sample
+          res = SHARED.keys.get_public_key(active_wid, role, id)
+          expect(res).to be_correct_and_respond 200
+          expect(res.to_s).to include addr_prefix
+        end
+
+        matrix_h.each do |role, addr_prefix|
+          id = [*0..100000].sample
+          res = SHARED.keys.get_public_key(active_wid, role, id, {hash: true})
+          expect(res).to be_correct_and_respond 200
+          expect(res.to_s).to include addr_prefix
+        end
+      end
+
+    end
+
+    describe "Account Public Keys" do
+      it "Create account public key - pending wallet from mnemonics" do
+        m24 = mnemonic_sentence(24)
+        acc_xpub = cardano_address_get_acc_xpub(m24, "1854H/1815H/0H")
+        pending_wid = create_pending_shared_wallet(m24, '0H', acc_xpub)
+        ["0H", "1H", "2147483647H", "44H"].each do |index|
+          res = SHARED.keys.create_acc_public_key(pending_wid, index, PASS, 'extended')
+          expect(res).to be_correct_and_respond 202
+          expect(res.to_s).to include "acct_shared_xvk"
+
+          res = SHARED.keys.create_acc_public_key(pending_wid, index, PASS, 'non_extended')
+          expect(res).to be_correct_and_respond 202
+          expect(res.to_s).to include "acct_shared_vk"
+        end
+      end
+
+      it "Create account public key - pending wallet from acc pub key" do
+        pending 'no_root key error on wallet from acc pub key'
```

Comment:
Also here for the fact that currently there is 'no_root key error' returned when trying to get acc pub key from wallets created from acc pub key.